### DISCUSSION TITLE
Using API to set folder for building GitHub Pages site

### DISCUSSION THREAD

**Discussion by shanjon** [community/community#57394] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I have a website hosted on GitHub Pages, and I would like to update the folder from which the site is being built. Currently, I do this via the GUI by navigating to **Settings** > **Pages** > **Build and deployment** > **Branch** and then selecting the folder (as shown here):

![image](https://github.com/community/community/assets/68360819/7ea317e6-2946-4b1c-b3c4-1d426046ea4b)

I am trying to determine if this can be done via API instead of via the GUI. I see in the GitHub Pages API that there is a call to '[Update information about a GitHub Pages site](https://docs.github.com/en/rest/pages?apiVersion=2022-11-28#update-information-about-a-github-pages-site)', but it is unclear to me if this one will allow me to set the folder for building the site. I see `path` in the example, but it is not listed in parameters.

**Comment by anhtran917** [comment, marked as answer]
Hi Shanjon,
To answer your question, it is possible to update the folder via the GitHub Pages API.

The API call that you mentioned, 'Update information about a GitHub Pages site', can be used to update the repository settings for your GitHub Pages site, including the source branch and folder. 
To update the folder, you can include the source parameter in your API call and set it to the path of the folder that you want to use for building your site.
I think you can refer  this code.
```
curl -X PATCH \
  -H "Authorization: token YOUR_TOKEN" \
  -H "Accept: application/vnd.github.switcheroo-preview+json" \
  https://api.github.com/repos/YOUR_USERNAME/YOUR_REPO/pages \
  -d '{"source": {"branch": "master", "path": "new_folder"}}'
```

Replace YOUR_TOKEN, YOUR_USERNAME, and YOUR_REPO with your own values. 
Also, replace new_folder with the name of the folder that you want to use for building your site.

I hope this helps!

Best regards, 
Anh

**Reply by shanjon** [reply]
Thank you so much!!!
